### PR TITLE
OSDOCS-471: Docs for backing up and restoring encryption keys

### PR DESCRIPTION
**Pull request by bergerhoffer**

### CODE REVIEW

**Comment by bergerhoffer** [comment]
@retroflexer Can you please review?

**Comment by bergerhoffer** [comment]
And @hexfusion too for good measure please :)

**Comment by bergerhoffer** [comment]
@openshift/team-documentation For peer review please

File: modules/dr-recover-lost-control-plane-hosts.adoc [hunk 1]
```diff
@@ -21,8 +21,19 @@ Follow these steps to recover from the loss of the majority of master hosts, lea
 +
 This procedure assumes that you have copied a snapshot file called `snapshot.db` to the `/home/core/` directory of your master host.
 
+.. If the etcd snapshot was encrypted, copy the encryption keys to the remaining master host.
++
+This procedure assumes that you have copied the `encryption-config` file to the `/home/core/` directory of your master host.
```
Comment (kalexand-rh) [inline modules/dr-recover-lost-control-plane-hosts.adoc]: ~have~ (both places)

File: modules/dr-restoring-cluster-state.adoc [hunk 2]
```diff
@@ -23,8 +23,19 @@ You should run the restore script on all of your master hosts within a short per
 +
 This procedure assumes that you have copied a snapshot file called `snapshot.db` to the `/home/core/` directory of your master host.
 
+.. If the etcd snapshot was encrypted, copy the encryption keys to the master host.
++
+This procedure assumes that you have copied the `encryption-config` file to the `/home/core/` directory of your master host.
+
 .. Access the master host.
 
+.. If the etcd snapshot was encrypted, restore the `encryption-config` file to the proper location.
```
Comment (kalexand-rh) [inline modules/dr-restoring-cluster-state.adoc]: s/proper/appropriate

> ! proper adj
Do not use to mean “correct,” “applicable,” or “appropriate.”

Is there more nuance than just "put it in the most recent revision of the `kube-apiserver-pod-<revision>` directory?" If not, consider ~to the proper location~ instead.

File: modules/enabling-etcd-encryption.adoc [hunk 3]
```diff
@@ -30,8 +30,9 @@ $ oc edit apiserver
 ----
 spec:
   encryption:
-    type: aescbc
+    type: aescbc <1>
 ----
+<1> The `aescbc` type means that AES-CBC with PKCS#7 padding and a 32 byte key is used to perform the encryption.
```
Comment (kalexand-rh) [inline modules/enabling-etcd-encryption.adoc]: I think that we need to point out that this is a FIPS approved cryptography package per https://github.com/openshift/openshift-docs/pull/18428#discussion_r356785590, but I will probably add it to that PR after you merge your PR.

**Review by kalexand-rh — COMMENTED** [review]
I have a few picks, but this looks good!

**Comment by bergerhoffer** [comment]
@geliu2016 Can you please review these docs for backing up/restoring from encrypted etcd? Thanks!

FYI @xingxingxia

File: modules/dr-recover-lost-control-plane-hosts.adoc [hunk 4]
```diff
@@ -31,10 +33,10 @@ This procedure assumes that you have copied a snapshot file called `snapshot.db`
 
 .. Run the `etcd-snapshot-restore.sh` script.
 +
-Pass in two parameters to the `etcd-snapshot-restore.sh` script: the path to the backed up etcd snapshot file and list of members, which is defined by the `INITIAL_CLUSTER` variable.
+Pass in two parameters to the `etcd-snapshot-restore.sh` script: the path to the etcd backup file and list of members, which is defined by the `INITIAL_CLUSTER` variable.
 +
 ----
-[core@ip-10-0-143-125 ~]$ sudo /usr/local/bin/etcd-snapshot-restore.sh /home/core/snapshot.db $INITIAL_CLUSTER
+[core@ip-10-0-143-125 ~]$ sudo /usr/local/bin/etcd-snapshot-restore.sh /home/core/snapshot_db_kuberesources_<datetimestamp>.tar.gz $INITIAL_CLUSTER
```
Comment (geliu2016) [inline modules/dr-recover-lost-control-plane-hosts.adoc]: Hello @retroflexer, Is the command line parameter changed from: 'snapshot.db' to 'snapshot_db_kuberesources_<datetimestamp>.tar.gz' ?  and 'snapshot_db_kuberesources_<datetimestamp>.tar.gz'  is zip file of directory: './asset', right?
if yes, is this change will be put in use from 4.3 only? or backport to 4.1,4.2?
pls correct me if there is any misunderstanding, thanks in advance!
FYI @bergerhoffer

Comment (retroflexer) [reply modules/dr-recover-lost-control-plane-hosts.adoc]: This 4.3 and beyond. Doesn't apply to 4.1 or 4.2. For backward compatibility, we allow specifying snapshot.db as well, if they have older backups (from 4.1 or 4.2).

**Pull request merged**